pkg/manager/getconfig.go

```diff
@@ -28,11 +28,24 @@ func (m *Manager) GetNetworkConfig(target string, configname string, path string
 	fmt.Println("Getting config for", target, path)
 	//TODO the key of the config store should be a tuple of (devicename, configname) use the param
 	var config store.Configuration
-	for configID, cfg := range m.ConfigStore.Store {
-		if cfg.Device == target {
-			configname = configID
-			config = cfg
-			break
+	if target != "" {
+		for _, cfg := range m.ConfigStore.Store {
+			if cfg.Device == target {
+				config = cfg
+				break
+			}
+		}
+		if config.Name == "" {
+			return make([]change.ConfigValue, 0),
+				fmt.Errorf("No Configuration found for %s", target)
+		}
```

Comment:
extract this if and use it at line 44 with else clause

Reply:
Currently we're only using the target name to find the Config name. I was leaving this here in case we use it differently in future.
I've added in the else stmt